src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -0,0 +1,222 @@
+}
+
+void CodeGen::genSSE3Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement SSE3 intrinsic code generation
+}
+
+void CodeGen::genSSSE3Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement SSSE3 intrinsic code generation
+}
+
+void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement SSE41 intrinsic code generation
+}
+
+void CodeGen::genSSE42Intrinsic(GenTreeHWIntrinsic* node)
+{
+    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    GenTree*       op1         = node->gtGetOp1();
+    GenTree*       op2         = node->gtGetOp2();
+    regNumber      targetReg   = node->gtRegNum;
+    assert(targetReg != REG_NA);
+    var_types targetType = node->TypeGet();
+    var_types baseType   = node->gtBaseType;
+
+    regNumber op1Reg = op1->gtRegNum;
+    regNumber op2Reg = op2->gtRegNum;
+    genConsumeOperands(node);
+
+    switch (intrinsicID)
+    {
+        case NI_SSE42_Crc32:
+            if (op1Reg != targetReg)
+            {
+                inst_RV_RV(INS_mov, targetReg, op1Reg, targetType, emitTypeSize(targetType));
+            }
+
+            if (baseType == TYP_UBYTE || baseType == TYP_CHAR) // baseType is the type of the second argument
+            {
+                assert(targetType == TYP_INT);
+                inst_RV_RV(INS_crc32, targetReg, op2Reg, baseType, emitTypeSize(baseType));
+            }
+            else
+            {
+                assert(op1->TypeGet() == op2->TypeGet());
+                assert(targetType == TYP_INT || targetType == TYP_LONG);
+                inst_RV_RV(INS_crc32, targetReg, op2Reg, targetType, emitTypeSize(targetType));
+            }
+
+            break;
+        default:
+            unreached();
+            break;
+    }
+    genProduceReg(node);
+}
+
+void CodeGen::genAVXIntrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement AVX intrinsic code generation
+}
+
+void CodeGen::genAVX2Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement AVX2 intrinsic code generation
+}
+
+void CodeGen::genAESIntrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement AES intrinsic code generation
+}
+
+void CodeGen::genBMI1Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement BMI1 intrinsic code generation
+}
+
+void CodeGen::genBMI2Intrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement BMI2 intrinsic code generation
+}
+
+void CodeGen::genFMAIntrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement FMA intrinsic code generation
+}
+
+void CodeGen::genLZCNTIntrinsic(GenTreeHWIntrinsic* node)
+{
+    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    GenTree*       op1         = node->gtGetOp1();
+    regNumber      targetReg   = node->gtRegNum;
+    assert(targetReg != REG_NA);
+    var_types targetType = node->TypeGet();
+    regNumber op1Reg     = op1->gtRegNum;
+    genConsumeOperands(node);
+
+    assert(intrinsicID == NI_LZCNT_LeadingZeroCount);
```

Comment:
Not a big issue (and I'm probably not consistent in this myself), but I don't think these kinds of asserts (in a method that is clearly called from a switch or conditional that is checking for this) are all that useful.